opensearchpy/connection/http_requests.py

```diff
@@ -188,7 +192,11 @@ def perform_request(  # type: ignore
         }
         send_kwargs.update(settings)
         try:
+            if self.metrics is not None:
+                self.metrics.request_start()
             response = self.session.send(prepared_request, **send_kwargs)
+            if self.metrics is not None:
```

Comment:
If there's an exception this is never called. Use `finally` to always `request_end`. Add a test.